lib/src/navigation.dart

```diff
@@ -151,9 +160,24 @@ class BottomNavScaffold extends ConsumerWidget {
   /// If the route is already at the first route, scroll the tab's root
   /// scrollable to the top.
   /// Otherwise, switch to the tapped tab.
-  void _onItemTapped(WidgetRef ref, int index) {
+  void _onItemTapped(
+    BuildContext context,
+    WidgetRef ref,
+    int index,
+    bool isOnline,
+  ) {
     final curTab = ref.read(currentBottomTabProvider);
     final tappedTab = BottomTab.values[index];
+
+    if (!isOnline && tappedTab == BottomTab.watch) {
```

Comment:
This logic would be better placed in the `else` of 
```dart
if (tappedTab == curTab) {
...
} else {
 // put it here because we want it to apply only if the tapped tab is not the current tab
}
```